src/interface/blas2/backend/default.hpp

```diff
@@ -137,8 +137,21 @@ typename sb_handle_t::event_t _trsv(
     sb_handle_t& sb_handle, index_t _N, container_t0 _mA, index_t _lda,
     container_t1 _vx, increment_t _incx,
     typename sb_handle_t::event_t _dependencies) {
-  return blas::internal::_trsv_impl<4, 2, uplo, trn, diag>(
-      sb_handle, _N, _mA, _lda, _vx, _incx, _dependencies);
+  const auto device = sb_handle.get_queue().get_device();
+  const std::string vendor =
+      device.template get_info<sycl::info::device::vendor>();
+  if (device.is_gpu()) {
+    if (vendor.find("Intel") == vendor.npos) {
+      return blas::internal::_trsv_impl<32, 4, uplo, trn, diag>(
+          sb_handle, _N, _mA, _lda, _vx, _incx, _dependencies);
+    } else {
```

Comment:
```suggestion
  if (device.is_gpu()) {
    const std::string vendor =
        device.template get_info<sycl::info::device::vendor>();
    if (vendor.find("Intel") == vendor.npos) {
      return blas::internal::_trsv_impl<32, 4, uplo, trn, diag>(
          sb_handle, _N, _mA, _lda, _vx, _incx, _dependencies);
    } else {
```

Reply:
Thank you @hjabird , pushed changes in 6a91a46eb187668a82aaf2f542fcabc6052c5eeb